collection.go

```diff
@@ -484,7 +483,7 @@ func (cl *collectionLoader) loadProgram(progName string) (*Program, error) {
 		if fd < 0 {
```

Comment:
`fd` is not necessary anymore?

Reply:
It is only used here to make sure that the passed map is valid. It returns a `sys.ErrClosedFd` which I guess holds meaning. 

Should we move this check to `Instruction.Marshal`? Or do you propose to not check map validity at all?

Reply:
Yeah `Instruction.Marshal` sounds like the place to do this.